### ISSUE TITLE
June 2019 Update (v1.36) breaks Arduino Extension Serial connection

### ISSUE THREAD

**Issue by aboulfad** [microsoft/vscode#76538]
- VSCode Version: 1.36
- OS Version: Mac Mojave 10.14.5

Steps to Reproduce:

1. Use v1.36, can not use serial connection anymore. See existing issue https://github.com/microsoft/vscode-arduino/issues/774
2. Revert to May update, issue is gone.

**Comment by dbaeumer** [comment]
@aboulfad do you see anything in the developer console. (e.g. Help -> Toggle Developer Tools).

**Comment by aboulfad** [comment]
Indeed, i switched back to 1.36 and noticed this when I clicked on the serial port in the bottom bar to select or change the serialport.

```
[Extension Host] [Warn] no library available after trying files 
(16) ["/Users/toto/.vscode/extensions/vsciot-vscode.vsco…ative/lib/native/usb-native_darwin_1.4.6_x64.node", "/Users/toto/.vscode/extensions/vsciot-vscode.vsco…ative/lib/native/usb-native_darwin_1.6.6_x64.node", "/Users/toto/.vscode/extensions/vsciot-vscode.vsco…ative/lib/native/usb-native_darwin_1.7.3_x64.node", "/Users/toto/.vscode/extensions/vsciot-vscode.vsco…ative/lib/native/usb-native_darwin_2.0.2_x64.node", "/Users/toto/.vscode/extensions/vsciot-vscode.vsco…tive/lib/native/usb-native_darwin_3.0.10_x64.node", "/Users/toto/.vscode/extensions/vsciot-vscode.vsco…/lib/native/usb-native_Ubuntu14.04_1.4.6_x64.node", "/Users/toto/.vscode/extensions/vsciot-vscode.vsco…/lib/native/usb-native_Ubuntu14.04_1.6.6_x64.node", "/Users/toto/.vscode/extensions/vsciot-vscode.vsco…/lib/native/usb-native_Ubuntu14.04_1.7.3_x64.node", "/Users/toto/.vscode/extensions/vsciot-vscode.vsco…/lib/native/usb-native_Ubuntu14.04_2.0.2_x64.node", "/Users/toto/.vscode/extensions/vsciot-vscode.vsco…lib/native/usb-native_Ubuntu14.04_3.0.10_x64.node", "/Users/toto/.vscode/extensions/vsciot-vscode.vsco…native/lib/native/usb-native_win32_1.4.6_x64.node", "/Users/toto/.vscode/extensions/vsciot-vscode.vsco…native/lib/native/usb-native_win32_1.4.7_x64.node", "/Users/toto/.vscode/extensions/vsciot-vscode.vsco…native/lib/native/usb-native_win32_1.6.6_x64.node", "/Users/toto/.vscode/extensions/vsciot-vscode.vsco…native/lib/native/usb-native_win32_1.7.3_x64.node", "/Users/toto/.vscode/extensions/vsciot-vscode.vsco…native/lib/native/usb-native_win32_2.0.2_x64.node", "/Users/toto/.vscode/extensions/vsciot-vscode.vsco…ative/lib/native/usb-native_win32_3.0.10_x64.node"]
0: "/Users/toto/.vscode/extensions/vsciot-vscode.vscode-arduino-0.2.26/vendor/node-usb-native/lib/native/usb-native_darwin_1.4.6_x64.node"
1: "/Users/toto/.vscode/extensions/vsciot-vscode.vscode-arduino-0.2.26/vendor/node-usb-native/lib/native/usb-native_darwin_1.6.6_x64.node"
2: "/Users/toto/.vscode/extensions/vsciot-vscode.vscode-arduino-0.2.26/vendor/node-usb-native/lib/native/usb-native_darwin_1.7.3_x64.node"
3: "/Users/toto/.vscode/extensions/vsciot-vscode.vscode-arduino-0.2.26/vendor/node-usb-native/lib/native/usb-native_darwin_2.0.2_x64.node"
4: "/Users/toto/.vscode/extensions/vsciot-vscode.vscode-arduino-0.2.26/vendor/node-usb-native/lib/native/usb-native_darwin_3.0.10_x64.node"
5: "/Users/toto/.vscode/extensions/vsciot-vscode.vscode-arduino-0.2.26/vendor/node-usb-native/lib/native/usb-native_Ubuntu14.04_1.4.6_x64.node"
6: "/Users/toto/.vscode/extensions/vsciot-vscode.vscode-arduino-0.2.26/vendor/node-usb-native/lib/native/usb-native_Ubuntu14.04_1.6.6_x64.node"
7: "/Users/toto/.vscode/extensions/vsciot-vscode.vscode-arduino-0.2.26/vendor/node-usb-native/lib/native/usb-native_Ubuntu14.04_1.7.3_x64.node"
8: "/Users/toto/.vscode/extensions/vsciot-vscode.vscode-arduino-0.2.26/vendor/node-usb-native/lib/native/usb-native_Ubuntu14.04_2.0.2_x64.node"
9: "/Users/toto/.vscode/extensions/vsciot-vscode.vscode-arduino-0.2.26/vendor/node-usb-native/lib/native/usb-native_Ubuntu14.04_3.0.10_x64.node"
10: "/Users/toto/.vscode/extensions/vsciot-vscode.vscode-arduino-0.2.26/vendor/node-usb-native/lib/native/usb-native_win32_1.4.6_x64.node"
11: "/Users/toto/.vscode/extensions/vsciot-vscode.vscode-arduino-0.2.26/vendor/node-usb-native/lib/native/usb-native_win32_1.4.7_x64.node"
12: "/Users/toto/.vscode/extensions/vsciot-vscode.vscode-arduino-0.2.26/vendor/node-usb-native/lib/native/usb-native_win32_1.6.6_x64.node"
13: "/Users/toto/.vscode/extensions/vsciot-vscode.vscode-arduino-0.2.26/vendor/node-usb-native/lib/native/usb-native_win32_1.7.3_x64.node"
14: "/Users/toto/.vscode/extensions/vsciot-vscode.vscode-arduino-0.2.26/vendor/node-usb-native/lib/native/usb-native_win32_2.0.2_x64.node"
15: "/Users/toto/.vscode/extensions/vsciot-vscode.vscode-arduino-0.2.26/vendor/node-usb-native/lib/native/usb-native_win32_3.0.10_x64.node"
length: 16```

**Comment by dbaeumer** [comment]
OK. This explains it. The problem is that the Arduino extension uses native node modules (which is not official supported right now in VS Code) and  the node version of VS Code has changed. This basically means that the Arduino folks need to ship a new version of their extension.

**Comment by aboulfad** [comment]
Thanks for the quick answer. I will open an issue on the vscode extension to follow up.

**Comment by ArthurMa1978** [comment]
Fixed since Arduino extension v0.2.27, thanks @aboulfad @dbaeumer
